fv3core/grid/generation.py

```diff
@@ -0,0 +1,1559 @@
+import functools
+
+import fv3gfs.util as fv3util
+from fv3core.utils.corners import (
+    fill_corners_2d,
+    fill_corners_agrid,
+    fill_corners_cgrid,
+    fill_corners_dgrid,
+)
+from fv3core.utils.global_constants import CARTESIAN_DIM, LON_OR_LAT_DIM, PI, RADIUS
+from fv3core.utils.grid import GridIndexing
+from fv3gfs.util.constants import N_HALO_DEFAULT
+
+from .eta import set_eta
+from .geometry import (
+    calc_unit_vector_south,
+    calc_unit_vector_west,
+    calculate_divg_del6,
+    calculate_grid_a,
+    calculate_grid_z,
+    calculate_l2c_vu,
+    calculate_supergrid_cos_sin,
+    calculate_trig_uv,
+    edge_factors,
+    efactor_a2c_v,
+    generate_xy_unit_vectors,
+    get_center_vector,
+    supergrid_corner_fix,
+    unit_vector_lonlat,
+)
+from .gnomonic import (
+    get_area,
+    great_circle_distance_along_axis,
+    local_gnomonic_ed,
+    lon_lat_corner_to_cell_center,
+    lon_lat_midpoint,
+    lon_lat_to_xyz,
+    set_c_grid_tile_border_area,
+    set_corner_area_to_triangle_area,
+    set_tile_border_dxc,
+    set_tile_border_dyc,
+)
+from .mirror import mirror_grid
+
+
+# TODO remove this when using python 3.8+ everywhere, it comes for free
+def cached_property(func):
+    cached = None
+
+    @property
+    @functools.wraps(func)
+    def wrapped(*args, **kwargs):
+        nonlocal cached
+        if cached is None:
+            cached = func(*args, **kwargs)
+        return cached
+
+    return wrapped
+
+
+# TODO
+# corners use sizer + partitioner rather than GridIndexer,
+# requires fv3core clls to corners know what to do
```

Comment:
Typo, but also I'm not sure what this means? Can you clarify it - what does it need to know how to do?
```suggestion
# requires fv3core calls to corners know what to do
```